picamctl/picamctl.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+# akseidel 02/06/2021
+# A script automating the steps to run showmewencam's "camera-ctl".
+# The script will pause for continue or exit if run with an argument.
+
+# edit this next line to be the correct text pattern for your computer.
+portnamepat="tty.usbmodem" #typical for showmewebcam on Appl OS X
+#portnamepat="ttyACM" #probably typical for showmewebcam on Linux
+#portnamepat="tty."
+# The client is an application you want this script to also run. 
+runtheclient="true"
+clientname="Photo Booth"
+
+# initial cleanup
+initclean(){
+    reset
+    clear
+    # terminates all the ort screen sessions.
+    screen -ls | grep Attached | cut -d. -f1 | awk '{print $1}' | xargs kill
+    screen -ls | grep Detached | cut -d. -f1 | awk '{print $1}' | xargs kill
+}
+
+# run the client application
+runclient(){
+    if [ "$runtheclient" = "true" ]; then
+        # -g causes application to open in background 
+        # https://scriptingosx.com/2017/02/the-macos-open-command/
+        open -g -a "$clientname"
```

Comment:
On Linux, this would probably be xdg-open

Reply:
Thanks. Hopefully I'll get the platform checking to work.

Reply:
In combination with auto discover the application Webcamoid is run for Linux situations. This has been tested on a Raspberry Pi 4 running a Raspbian system.

I could be wrong but xdg-open wants an image file for an argument, so I did not find a way to use it. In the script situation an application needs to be run, For the Linux situation I found running a detached  screen running Webcamoid as a solution. The difference between the Apple and Linux situations is this detached screen existence that gets blown away the next time the script is run. Webcamoid closes and reopens if it was already running whereas an already running PhotoBooth just stays.